core/src/test/java/org/kiwix/kiwixmobile/core/dao/NewBookmarksDaoTest.kt

```diff
@@ -0,0 +1,162 @@
+    every { queryBuilder.build() } returns query
+    newBookmarksDao.getBookmarks(false, zimFileReader)
+    verify { queryBuilder.build() }
+  }
+
+  @Test
+  fun `get bookmarks from current book`() {
+    mockkStatic(CoreApp::class)
+    every { CoreApp.getInstance().packageName } returns "pkg"
+    val zimFileReader: ZimFileReader = mockk()
+    val queryBuilder: QueryBuilder<BookmarkEntity> = mockk()
+    every { box.query() } returns queryBuilder
+    every { zimFileReader.name } returns ""
+    every {
+      queryBuilder.equal(
+        BookmarkEntity_.zimName,
+        zimFileReader.name
+      )
+    } returns queryBuilder
+    every { queryBuilder.order(BookmarkEntity_.bookmarkTitle) } returns queryBuilder
+    val query: Query<BookmarkEntity> = mockk(relaxed = true)
+    every { queryBuilder.build() } returns query
+    newBookmarksDao.getBookmarks(true, zimFileReader)
+    verify { queryBuilder.equal(BookmarkEntity_.zimName, zimFileReader.name) }
+    verify { queryBuilder.build() }
+  }
+
+  @Test
+  fun `get current zim bookmarks url`() {
+    mockkStatic(CoreApp::class)
+    every { CoreApp.getInstance().packageName } returns "pkg"
+    val zimFileReader: ZimFileReader = mockk()
+    val queryBuilder: QueryBuilder<BookmarkEntity> = mockk()
+    every { box.query() } returns queryBuilder
+    every { zimFileReader.name } returns ""
+    every {
+      queryBuilder.equal(
+        BookmarkEntity_.zimName,
+        zimFileReader.name
+      )
+    } returns queryBuilder
+    every { queryBuilder.or() } returns queryBuilder
+    every { zimFileReader.id } returns ""
+    every {
+      queryBuilder.equal(
+        BookmarkEntity_.zimId,
+        zimFileReader.id
+      )
+    } returns queryBuilder
+    every { queryBuilder.order(BookmarkEntity_.bookmarkTitle) } returns queryBuilder
+  }
+
+  @Test
+  fun `bookmark urls for current workbook`() {
+    mockkStatic(CoreApp::class)
+    every { CoreApp.getInstance().packageName } returns "pkg"
+    val zimFileReader: ZimFileReader = mockk()
+    val queryBuilder: QueryBuilder<BookmarkEntity> = mockk()
+    every { box.query() } returns queryBuilder
+    every { zimFileReader.name } returns ""
+    every {
+      queryBuilder.equal(
+        BookmarkEntity_.zimName,
+        zimFileReader.name
+      )
+    } returns queryBuilder
+    every { queryBuilder.or() } returns queryBuilder
+    every { zimFileReader.id } returns ""
+    every {
+      queryBuilder.equal(
+        BookmarkEntity_.zimId,
+        zimFileReader.id
+      )
+    } returns queryBuilder
+    every { queryBuilder.order(BookmarkEntity_.bookmarkTitle) } returns queryBuilder
+    val query: Query<BookmarkEntity> = mockk(relaxed = true)
+    every { queryBuilder.build() } returns query
+    val queryResult: List<BookmarkEntity> = mockk()
+    mockkStatic(RxQuery::class)
+    every { RxQuery.observable(query) } returns Observable.just(queryResult)
+    newBookmarksDao.bookmarkUrlsForCurrentBook(zimFileReader)
```

Comment:
for things that return flowables you call `.test()` on them and then you get access to the assert function on a flowable.